code/modules/stock_market/stockmarket.dm

```diff
@@ -1,171 +1,171 @@
-		S.optimism = S.average_optimism + (rand(-40, 40) / 100)
-		S.current_trend = rand(-200, 200) / 10
-		S.last_trend = S.current_trend
-		S.disp_value_change = rand(-1, 1)
-		S.speculation = rand(-20, 20)
-		S.average_shares = round(rand(500, 10000) / 10)
-		S.outside_shareholders = rand(1000, 30000)
-		S.available_shares = rand(200000, 800000)
-		S.fluctuation_rate = rand(6, 20)
-		S.generateIndustry()
-		S.generateEvents()
-		stocks += S
-		last_read[S] = list()
-
-/datum/stockMarket/process()
-	for (var/stock in stocks)
-		var/datum/stock/S = stock
-		S.process()
-
-/datum/stockMarket/proc/add_log(var/log_type, var/user, var/company_name, var/stocks, var/shareprice, var/money)
-	var/datum/stock_log/L = new log_type
-	L.user_name = user
-	L.company_name = company_name
-	L.stocks = stocks
-	L.shareprice = shareprice
-	L.money = money
-	L.time = time2text(world.timeofday, "hh:mm")
-	logs += L
-
-var/global/datum/stockMarket/stockExchange = new
-
-/proc/plotBarGraph(var/list/points, var/base_text, var/width=400, var/height=400)
-	var/output = "<table style='border:1px solid black; border-collapse: collapse; width: [width]px; height: [height]px'>"
-	if (points.len && height > 20 && width > 20)
-		var/min = points[1]
-		var/max = points[1]
-		for (var/v in points)
-			if (v < min)
-				min = v
-			if (v > max)
-				max = v
-		var/cells = (height - 20) / 20
-		if (cells > round(cells))
-			cells = round(cells) + 1
-		var/diff = max - min
-		var/ost = diff / cells
-		if (min > 0)
-			min = max(min - ost, 0)
-		diff = max - min
-		ost = diff / cells
-		var/cval = max
-		var/cwid = width / (points.len + 1)
-		for (var/y = cells, y > 0, y--)
-			if (y == cells)
-				output += "<tr>"
-			else
-				output += "<tr style='border:none; border-top:1px solid #00ff00; height: 20px'>"
-			for (var/x = 0, x <= points.len, x++)
-				if (x == 0)
-					output += "<td style='border:none; height: 20px; width: [cwid]px; font-size:10px; color:#00ff00; background:black; text-align:right; vertical-align:bottom'>[round(cval - ost)]</td>"
-				else
-					var/v = points[x]
-					if (v >= cval)
-						output += "<td style='border:none; height: 20px; width: [cwid]px; background:#0000ff'>&nbsp;</td>"
-					else
-						output += "<td style='border:none; height: 20px; width: [cwid]px; background:black'>&nbsp;</td>"
-			output += "</tr>"
-			cval -= ost
-		output += "<tr><td style='font-size:10px; height: 20px; width: 100%; background:black; color:green; text-align:center' colspan='[points.len + 1]'>[base_text]</td></tr>"
-	else
-		output += "<tr><td style='width:[width]px; height:[height]px; background: black'></td></tr>"
-		output += "<tr><td style='font-size:10px; background:black; color:green; text-align:center'>[base_text]</td></tr>"
-
-	return "[output]</table>"
-
+
+
+/datum/stockMarket
+	var/list/stocks = list()
+	var/list/balances = list()
+	var/list/last_read = list()
+	var/list/stockBrokers = list()
+	var/list/logs = list()
+
+/datum/stockMarket/New()
+		..()

```

Comment:
no idea. also, goof

Reminder that when you comment on a line, you should comment below all off the related lines, other wise only one line shows.

your comment should have been 4 or 5 lines down